readthedocs/doc_builder/python_environments.py

```diff
@@ -337,21 +337,25 @@ def install_core_requirements(self):
             *self._pip_cache_cmd_argument(),
         ]
 
-        # Install latest pip first,
+        # Install latest pip and setuptools first,
         # so it is used when installing the other requirements.
         pip_version = self.project.get_feature_value(
             Feature.DONT_INSTALL_LATEST_PIP,
             # 20.3 uses the new resolver by default.
             positive='pip<20.3',
             negative='pip',
         )
-        cmd = pip_install_cmd + [pip_version]
+        setuptools_version = self.project.get_feature_value(
+            Feature.INSTALL_LATEST_SETUPTOOLS,
+            positive='setuptools',
+            negative='setuptools==41.0.1',
+        )
+        cmd = pip_install_cmd + [pip_version, setuptools_version]
         self.build_env.run(
             *cmd, bin_path=self.venv_bin(), cwd=self.checkout_path
         )
```

Comment:
We are changing _where_ we install `setuptools` here.

Now we are installing it together with `pip` as the first command we executed. Before, we were installing it together with our own dependencies. I don't think this is something bad, but definitely a change that may break in a random way :upside_down_face: 

It makes sense to be installed with `pip` to me; but we should keep an eye on this.